diff --git a/Cargo.toml b/Cargo.toml
index b10c546..8d5bcb4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,6 +14,7 @@ categories = ["gui", "multimedia"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+libc = "0.2"
 pipewire = "0.7.1"
 adw = { version = "0.5", package = "libadwaita", features = ["v1_3"] }
 glib = { version = "0.18", features = ["log"] }
@@ -22,4 +23,3 @@ log = "0.4.11"
 
 once_cell = "1.7.2"
 
-libc = "0.2"
