diff --git a/Cargo.toml b/Cargo.toml
index b10c546..428c4d0 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,7 +14,7 @@ categories = ["gui", "multimedia"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-pipewire = "0.7.1"
+pipewire = "0.8.0"
 adw = { version = "0.5", package = "libadwaita", features = ["v1_3"] }
 glib = { version = "0.18", features = ["log"] }
 
